indexer/src/main/java/juice/modules/ConsumerModule.java

```diff
@@ -0,0 +1,34 @@
+package juice.modules;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import config.ConfigFileFinder;
+import config.ConfigurationFactory;
+import config.ConsumerConfiguration;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+
+import java.util.Properties;
+
+public class ConsumerModule extends AbstractModule {
+
+    private final String CONFIG_FILE_NAME = "consumer.config";
+
+
+    @Provides
+    public KafkaConsumer<String, String> provideConsumer() {
```

Comment:
In the case of the `KafkaConsumer`, I'm not sure we want to bind it using Guice. The question you should ask your self is - Would the consumer be used by multiple classes? If no, you can simply create it in the class that is using it

Reply:
why does it matter if we inject it or construct it in IndexerConsumer? (the class that is in charge of message consumption, thus has a KafkaConsumer member which is initialised by injection). More generally, what is the advantage of regular initialisation over guice injection?